### ISSUE TITLE
[bug]: labelFormatter in ChartTooltip component not working properly when label type is not string

### ISSUE THREAD

**Issue by tonghohin** [shadcn-ui/ui#4208]
### Describe the bug

I'm trying to build a chart like the Line Chart - Interactive on this page: https://ui.shadcn.com/charts#line-chart, with date shown in each tooltip. But it doesn't work and when I go to Edit in V0, it's also showing invalid date on the tooltips.

### Affected component/components

ChartTooltip

### How to reproduce

1. Go to https://v0.dev/r/Gv5F7HkGIGo
2. Hover on the chart and you'll see invalid date on the tooltips.

### Codesandbox/StackBlitz link

https://v0.dev/r/Gv5F7HkGIGo

### System Info

```bash
Safari
```

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues

**Comment by tonghohin** [comment]
I fixed it by removing '&& typeof label === "string"' at line 112 in chart.tsx

**Comment by shehbajdhillon** [comment]
> I fixed it by removing '&& typeof label === "string"' at line 112 in chart.tsx

Ran into the same problem and this fixed it for me thank u so much

**Comment by joeychrys** [comment]
> > I fixed it by removing '&& typeof label === "string"' at line 112 in chart.tsx
> 
> Ran into the same problem and this fixed it for me thank u so much

Noticed this too :/ but at least its an easy fix

**Comment by dr-msr** [comment]
It is now at line 145

**Comment by EnzoPV25** [comment]
Hi, same problem fixed the same way in my case. 

I guess this shouldn't be closed since the problem is still there

**Comment by reZerOR** [comment]
> I fixed it by removing '&& typeof label === "string"' at line 112 in chart.tsx

thanks

**Comment by tonghohin** [comment]
Reopen the issue as I just ran into this bug again when I pulled the latest Chart component. And it's now at line 127.

**Comment by DiAifU** [comment]
Yep, having the same need ! Removing the typeof string part line 145 fixes the problem as mentionned by @tonghohin !

**Comment by shadcn** [comment]
This is now fixed in latest. Thank you.
